src/components/molecules/UserProfilePicture/UserProfilePicture.tsx

```diff
@@ -16,154 +17,176 @@ import {
 
 import { User } from "types/User";
 
-// @debt remove styled-components in favour of using our standard scss patterns
-import * as S from "./UserProfilePicture.styles";
 import "./UserProfilePicture.scss";
 
+const randomAvatarUrl = (id: string) =>
+  "/avatars/" +
+  RANDOM_AVATARS[Math.floor(id?.charCodeAt(0) % RANDOM_AVATARS.length)];
+
+// TODO: refactor this to accept named props instead of positional args
+const avatarUrl = (
+  id: string,
+  anonMode?: boolean,
+  pictureUrl?: string,
+  miniAvatars?: boolean
+): string => {
+  if (anonMode || !id) {
+    return DEFAULT_PROFILE_IMAGE;
+  }
+
+  if (!miniAvatars && pictureUrl) {
+    return pictureUrl;
+  }
+
+  // TODO: how is this intended to be used? Why doesn't it use the profile image? It seems to be something set on venue config..
+  if (miniAvatars) {
+    return randomAvatarUrl(id);
+  }
+
+  return DEFAULT_PROFILE_IMAGE;
+};
+
 export interface UserProfilePictureProp {
   user: WithId<User>;
   isAudioEffectDisabled?: boolean;
   miniAvatars?: boolean;
   avatarClassName?: string;
   avatarStyle?: object;
   containerStyle?: object;
-  reactionPosition?: "right" | "left" | undefined;
+  reactionPosition?: "left" | "right";
 }
 
 // @debt This component should be divided into a few with simpler logic. Also, remove `styled components`
 // @debt the UserAvatar component serves a very similar purpose to this, we should unify them as much as possible
-const UserProfilePicture: React.FC<UserProfilePictureProp> = ({
+export const UserProfilePicture: React.FC<UserProfilePictureProp> = ({
   isAudioEffectDisabled = true,
   miniAvatars = false,
-  avatarClassName = "profile-icon",
-  avatarStyle,
-  containerStyle,
-  reactionPosition,
+  avatarClassName = "UserProfilePicture__avatar",
+  avatarStyle, // TODO: do we need this prop? Can we remove it? Or at least rename it? Only seems to be used in MapPartygoersOverlay
+  containerStyle, // TODO: do we need this prop? Can we remove it? Or at least rename it? Only seems to be used in MapPartygoersOverlay
```

Comment:
[TODO] do we need this prop? Can we remove it? Or at least rename it? Only seems to be used in `MapPartygoersOverlay`

Reply:
Potentially OOS

Reply:
fixed in https://github.com/sparkletown/sparkle/pull/1323/commits/a6a5eed78c56d517a3f5c00380a5ee775f37e97f + https://github.com/sparkletown/sparkle/pull/1323/commits/134a48d8abe7e0a93cf6e9045b5c6bc2d57f10a5 + https://github.com/sparkletown/sparkle/pull/1323/commits/0e71fc5bc49b5bbc0ecbaed763eb0d884c640374 + https://github.com/sparkletown/sparkle/pull/1323/commits/59073433104b9051729d8a4b6d83957c43fa4f76